pkg/networkserver/redis/application_uplink_queue.go

```diff
@@ -61,9 +69,26 @@ func (q *ApplicationUplinkQueue) Add(ctx context.Context, ups ...*ttnpb.Applicat
 			return err
 		}
 
+		var (
+			streamID             string
+			maxLen, maxLenApprox int64
+		)
+		switch up.Up.(type) {
+		case *ttnpb.ApplicationUp_JoinAccept:
+			streamID = q.uidJoinAcceptKey(uid)
+			maxLenApprox = q.maxLen
+		case *ttnpb.ApplicationUp_DownlinkQueueInvalidated:
```

Comment:
I don't think this is ok. We cannot coalesce this per application, since we may have multiple devices in the same app.

Reply:
Ah yes indeed, good point!